### PR TITLE
mattermost: split into mattermost-server and mattermost-webapp

### PR DESCRIPTION
**Pull request by ryantm**
###### Motivation for this change

This improves overriding and build debug and paves the way for
attempting to build mattermost-webapp from source.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/mattermost/default.nix [hunk 1]
```diff
@@ -1,58 +1,14 @@
...
-    doCheck = false;
-
-    ldflags = [
-      "-s" "-w" "-X github.com/mattermost/mattermost-server/v${lib.versions.major version}/model.BuildNumber=${version}"
-    ];
-
+{ lib, mattermost-server, mattermost-webapp, buildEnv }:
+
+buildEnv {
+  name = "mattermost-${mattermost-server.version}";
```
Comment (SuperSandro2000) [inline pkgs/servers/mattermost/default.nix:4]: Does buildEnv not take pname+version?

File: pkgs/servers/mattermost/webapp/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  installPhase = ''
+    mkdir -p $out
+    tar --strip 1 --directory $out -xf $src \
+    mattermost/client \
+    mattermost/i18n \
+    mattermost/fonts \
+    mattermost/templates \
+    mattermost/config
```
Comment (SuperSandro2000) [inline pkgs/servers/mattermost/webapp/default.nix:19]:
```suggestion
      mattermost/client \
      mattermost/i18n \
      mattermost/fonts \
      mattermost/templates \
      mattermost/config
```

File: pkgs/servers/mattermost/webapp/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "Open-source, self-hosted Slack-alternative";
+    homepage = "https://www.mattermost.org";
+    license = with licenses; [ agpl3 asl20 ];
+    maintainers = with maintainers; [ fpletz ryantm ];
+    platforms = platforms.unix;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/servers/mattermost/webapp/default.nix:30]:
```suggestion
  };
}
```

**Comment by ryantm** [comment]
This is outdated.

**Pull request closed without merging**